gobblin-config-management/gobblin-config-client/src/main/java/gobblin/config/client/ConfigClientUtils.java

```diff
@@ -0,0 +1,176 @@
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.config.client;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.apache.hadoop.fs.Path;
+
+import com.google.common.base.Preconditions;
+
+import gobblin.config.common.impl.SingleLinkedListConfigKeyPath;
+import gobblin.config.store.api.ConfigKeyPath;
+import gobblin.config.store.api.ConfigStore;
+
+/**
+ * Utility class to transfer {@link URI} to {@link ConfigKeyPath} and vice versa 
+ * 
+ * @author mitu
+ *
+ */
+public class ConfigClientUtils {
+
+  /**
+   * 
+   * @param configKeyURI - URI provided by client , which could missing authority/store root directory
+   * @param cs           - ConfigStore corresponding to the input URI. Require input URI's scheme/authority name 
+   *                       match ConfigStore's scheme/authority
+   * @return             - {@link ConfigKeyPath} for the relative path
+   */
+  public static ConfigKeyPath buildConfigKeyPath(URI configKeyURI, ConfigStore cs){
+    checkMatchingSchemeAndAuthority(configKeyURI, cs);
+    // Example store root is   etl-hdfs://eat1-nertznn01.grid.linkedin.com:9000/user/mitu/HdfsBasedConfigTest
+    
+    // configKeyURI is etl-hdfs:///datasets/a1/a2
+    if(configKeyURI.getAuthority()==null){
+      return getConfigKeyPath(configKeyURI.getPath());
+    }
+    // configKeyURI is etl-hdfs://eat1-nertznn01.grid.linkedin.com:9000/user/mitu/HdfsBasedConfigTest/datasets/a1/a2
+    else {
+      URI relative = cs.getStoreURI().relativize(configKeyURI);
+      return getConfigKeyPath(relative.getPath());
+    } 
+  }
+  
+  /**
+   * Build the URI based on the {@link ConfigStore} or input cnofigKeyURI
+   * 
+   * @param configKeyPath : relative path to the input config store cs
+   * @param configKeyURI  : normally is the URI pass in by client application, could be
+   * 1. etl-hdfs:///datasets/a1/a2 which missing the authority
+   * 2. etl-hdfs://eat1-nertznn01.grid.linkedin.com:9000/user/mitu/HdfsBasedConfigTest/datasets/a1/a2 which has the authority
+   * @param cs            : the config store of the input configKeyURI
+   * @return              : return the URI of the same format with the input configKeyURI
+   * 
+   * for example, configKeyPath is /tags/retention, 
+   * with configKeyURI as etl-hdfs:///datasets/a1/a2, return "etl-hdfs:///tags/retention
+   * with configKeyURI as etl-hdfs://eat1-nertznn01.grid.linkedin.com:9000/user/mitu/HdfsBasedConfigTest , then return
+   * etl-hdfs://eat1-nertznn01.grid.linkedin.com:9000/user/mitu/HdfsBasedConfigTest/tags/retention
+   */
+  public static URI buildURI(ConfigKeyPath configKeyPath, URI configKeyURI, ConfigStore cs){
```

Comment:
I don't understand what this method is supposed to do. Looking at the two examples: in the first case, the path replaces the path in the URI; in the second case, the path is appended. Why do those two cases have a completely different treatment?